versioned_docs/version-V3/guides/providing-liquidity/setting-up-your-contract.md

```diff
@@ -0,0 +1,188 @@
+Every [NFT](https://eips.ethereum.org/EIPS/eip-721) is identified by a unique uint256 ID inside the ERC-721 smart contract, declared as the `tokenId`
+
+To allow deposits of ERC721 expressions of liquidity, we'll need to create a struct called `Deposit`, a mapping of `uint256` to the `Deposit` struct, then declare that mapping as a public variable `deposits`.
+
+```solidity
+    struct Deposit {
+        address owner;
+        uint128 liquidity;
+        address token0;
+        address token1;
+    }
+
+    mapping(uint256 => Deposit) public deposits;
+```
+
+```solidity
+    constructor(
+        INonfungiblePositionManager _nonfungiblePositionManager,
+        address _factory,
+        address _WETH9
+    ) PeripheryImmutableState(_factory, _WETH9) {
+        nonfungiblePositionManager = _nonfungiblePositionManager;
+    }
+```
+
+## The Constructor
+
+We'll declare the constructor here, which is executed once when the contract is deployed. Our constructor hard codes the address of the non fungible position manager interface, V3 router, and the Periphery immutable state constructor, which requires the factory, the address of weth9 (the [ERC-20 wrapper](https://weth.io/) for ether).
+
+```solidity
+   constructor(
+        INonfungiblePositionManager _nonfungiblePositionManager,
+        address _factory,
+        address _WETH9
+    ) PeripheryImmutableState(_factory, _WETH9) {
+        nonfungiblePositionManager = _nonfungiblePositionManager;
+    }
+```
+
+## Allowing custody of ERC721 tokens
+
+To allow our contract to custody ERC721 tokens - we'll implement the `onERC721Received` function within our inherited `IERC721Receiver.sol` [contract](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/IERC721Receiver.sol).
+
+because we are not using the `from` identifier, we can omit it in our arguments.
+
+```solidity
+    function onERC721Received(
+        address operator,
+        address,
+        uint256 tokenId,
+        bytes calldata
+    ) external override returns (bytes4) {
+        // get position information
+        _createDeposit(operator, tokenId);
+        return this.onERC721Received.selector;
+    }
+```
+
+## Creating a Deposit
+
+To add a `Deposit` instance to the `deposits` mapping, we'll create an internal function called `_createDeposit` that destructures the `positions` struct we have inherited from `nonfungiblePositionManager.sol`, and pass the relevant variables `token0` `token1` and `liquidity` to our `deposits` mapping.
+
+```solidity
+    function _createDeposit(address owner, uint256 tokenId) internal {
+        (, , address token0, address token1, , , , uint128 liquidity, , , , ) =
+            nonfungiblePositionManager.positions(tokenId);
+
+        // set the owner and data for position
+        // operator is msg.sender
+        deposits[tokenId] = Deposit({owner: owner, liquidity: liquidity, token0: token0, token1: token1});
+    }
+
+```
+
+## The Full Contract Setup
```

Comment:
will just have to edit this portion since it changed from the review :)

Reply:
not sure what you mean here - which part needs to be edited? hard to track what has changed